jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2982,7 +2982,7 @@ private static final MethodHandle foldArgumentsCommon(MethodHandle handle, int f
 		final int argIndexCount = argumentIndices.length;
 		int[] passedInArgumentIndices = EMPTY_ARG_POSITIONS;
 		
-		if ((foldPosition < 0) || (foldPosition >= handleTypeParamCount)) {
+		if ((foldPosition < 0) || (foldPosition > handleTypeParamCount) || ((handleTypeParamCount != 0) && (foldPosition == handleTypeParamCount))) {
```

Comment:
```suggestion
		if ((foldPosition < 0) 
		|| (foldPosition > handleTypeParamCount) 
		/* Note, folding at handleTypeParamCount is only valid when the handle takes no arguments */
		|| ((foldPosition == handleTypeParamCount) && (foldPosition != 0)))
		{
```
Can this be reformatted similar to this?  It makes the code clearer to be able to see the three cases separately.